ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/TransactionLogsIndexer.java

```diff
@@ -0,0 +1,187 @@
+    try {
+      indexingStatus.indexing = true;
+      LOG.info(
+          "Generating transaction log indexes from block {} to block {} in {}",
+          start,
+          stop,
+          cacheDir);
+      if (!Files.isDirectory(cacheDir) && !cacheDir.toFile().mkdirs()) {
+        LOG.error("Cache directory '{}' does not exist and could not be made.", cacheDir);
+        return indexingStatus;
+      }
+
+      final File pendingFile = calculateCacheFileName(PENDING, cacheDir);
+      for (long blockNum = start; blockNum < stop; blockNum += BLOCKS_PER_BLOOM_CACHE) {
+        LOG.info("Indexing segment at {}", blockNum);
+        try (final FileOutputStream fos = new FileOutputStream(pendingFile)) {
+          final long blockCount = fillCacheFile(blockNum, blockNum + BLOCKS_PER_BLOOM_CACHE, fos);
+          if (blockCount == BLOCKS_PER_BLOOM_CACHE) {
+            Files.move(
+                pendingFile.toPath(),
+                calculateCacheFileName(blockNum, cacheDir).toPath(),
+                StandardCopyOption.REPLACE_EXISTING,
+                StandardCopyOption.ATOMIC_MOVE);
+          } else {
+            LOG.info("Partial segment at {}, only {} blocks cached", blockNum, blockCount);
+            break;
+          }
+        }
+      }
+    } catch (final Exception e) {
+      LOG.error("Unhandled indexing exception", e);
+    } finally {
+      indexingStatus.indexing = false;
+      LOG.info("Indexing request complete");
+    }
+    return indexingStatus;
+  }
+
+  private long fillCacheFile(
+      final long startBlock, final long stopBlock, final FileOutputStream fos) throws IOException {
+    long blockNum = startBlock;
+    while (blockNum < stopBlock) {
+      final Optional<BlockHeader> maybeHeader = blockchain.getBlockHeader(blockNum);
+      if (maybeHeader.isEmpty()) {
+        break;
+      }
+      final byte[] logs = maybeHeader.get().getLogsBloom().getByteArray();
+      checkNotNull(logs);
+      checkState(logs.length == 256, "BloomBits are not the correct length");
+      fos.write(logs);
+      indexingStatus.currentBlock = blockNum;
+      blockNum++;
+    }
+    return blockNum - startBlock;
+  }
+
+  public IndexingStatus requestIndexing(final long fromBlock, final long toBlock) {
+    boolean requestAccepted = false;
+    try {
+      if ((fromBlock < toBlock) && submissionLock.tryLock(100, TimeUnit.MILLISECONDS)) {
+        try {
+          if (!indexingStatus.indexing) {
+            requestAccepted = true;
+            indexingStatus.startBlock = fromBlock;
+            indexingStatus.endBlock = toBlock;
+            scheduler.scheduleComputationTask(
+                () ->
+                    generateLogBloomCache(
+                        fromBlock - (fromBlock % BLOCKS_PER_BLOOM_CACHE), toBlock));
+          }
+        } finally {
+          submissionLock.unlock();
+        }
+      }
+    } catch (final InterruptedException e) {
+      // ignore
```

Comment:
nit; might a `DEBUG` or `TRACE` level log message be useful in this `InterruptedException` case, given it is around indexing?

Reply:
The only place it would be seen is when attempting to acquire the lock, and we wouldn't re-try if we got interrupted waiting for the lock, we would fail just like a timeout occured.  So I don't see logging as adding any value.